src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -894,7 +894,15 @@ protected void ParseSshHostName(string hostname, out string host, out string use
             try
             {
                 Uri uri = new System.Uri("ssh://" + hostname);
-                host = ResolveComputerName(uri.Host);
+
+                // Extract original host from URI with preserved case
+                // This is needed since System.Uri canonicalizes URI and makes host lowercase
+                int originalHostIndex = uri.OriginalString.IndexOf(uri.Host, StringComparison.OrdinalIgnoreCase);
```

Comment:
You'll probably need to investigate how/if IPv6 addresses can be used today. I know Uri will canonicalize the IPv6 address so the length of `uri.Host` won't match up with the original string.

It could also be that `-HostName ...` doesn't even support IPv6 addresses because I believe `ssh` doesn't suppor the `[::1]` notation that `Uri` might use.

Reply:
Evaluate `originalHostIndex` only if `uri.HostNameType == UriHostNameType.Dns`

Reply:
Thanks @jborean93 and @iSazonov. 

For now, I have added the condition to check DNS hostname and isolated the original host code in there. I'm honestly not an expert with IPv6 canonicalization nor have much background/experience if it ever worked in PowerShell. Probably need someone else with more knowledge in that area to comment. Maybe it's time I should become one because I keep hearing more and more, we should use IPv6 😄.

From [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=net-8.0):

_The URI class supports the use of IP addresses in both quad notation for the IPv4 protocol and colon hexadecimal for the IPv6 protocol. Enclose the IPv6 address in square brackets, for example, http://[::1]._

Doing a quick test it seems like `Host` is not affected by canonicalization:

```pwsh
> [System.Uri]::new("ssh://user@[2607:f8b0:4009:816::200e]")

AbsolutePath   : /
AbsoluteUri    : ssh://user@[2607:f8b0:4009:816::200e]/
LocalPath      : /
Authority      : [2607:f8b0:4009:816::200e]
HostNameType   : IPv6
IsDefaultPort  : True
IsFile         : False
IsLoopback     : False
PathAndQuery   : /
Segments       : {/}
IsUnc          : False
Host           : [2607:f8b0:4009:816::200e]
Port           : -1
Query          :
Fragment       :
Scheme         : ssh
OriginalString : ssh://user@[2607:f8b0:4009:816::200e]
DnsSafeHost    : 2607:f8b0:4009:816::200e
IdnHost        : 2607:f8b0:4009:816::200e
IsAbsoluteUri  : True
UserEscaped    : False
UserInfo       : user
```

And since we're now excluding it from the substring code for DNS hostnames it should be fine and won't break existing behaviour.

Reply:
Is user name case-sensitive too?

Reply:
> I'm honestly not an expert with IPv6 canonicalization nor have much background/experience if it ever worked in PowerShell. 

I don't think IPv6 ever works and honestly I wouldn't hold up this PR for that. It should probably be investigated separately and have this focus solely on ensuring the hostname stays the same.

> Is user name case-sensitive too?

The uri class won't lower case the username so we don't have to worry about it.